leankyr/Warmup_Problems/Utopian_Tree/Utopian_Tree.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3.5
+from __future__ import print_function
+
+import math
+import os
+import random
+import re
+import sys
+import logging
+#import ipdb
+
+logging.basicConfig(level=logging.WARN, format='%(levelname)s- %(message)s')
+
+
+
+def read_file(filename):
```

Comment:
Instead of opening a file like this it is better to use open with the `with` keyword which basically creates a [context manager](https://docs.python.org/3/library/contextlib.html) that will close the file at the end of the statement automatically
e.g.
```python
with open(os.environ[...], 'w') as fptr:
     # do stuff with the fptr in this scope
# when you reach here, you are out of scope and file is closed
```